packages/rich-text/src/component/index.native.js

```diff
@@ -817,6 +817,21 @@ export class RichText extends Component {
 				} else if ( this.props.selectionStart > record.text.length || this.props.selectionEnd > record.text.length ) {
 					console.warn( 'Oops, selection will land outside the text, skipping setting it...' );
 					selection = null;
+				} else if ( html.match( /(\s*<br[^>]*>\s*)+<\/p>$/ ) ) {
+					console.warn( 'Oops, BR tag(s) at the end of content. Aztec will remove them, adapting the selection...' );
+					const brBeforeParaMatches = html.match( /(\s*<br[^>]*>\s*)+<\/p>$/g );
```

Comment:
On a different thing, I notice that the regex is different than the one in Aztec ([here](https://github.com/wordpress-mobile/AztecEditor-Android/blob/b1fad439d56fa6d4aa0b78526fef355c59d00dd3/aztec/src/main/kotlin/org/wordpress/aztec/AztecParser.kt#L656)), and I wonder how this one came up. Especially, the `[^>]` part which seems to me that it would match tags like `<broken>` that have extra chars after the `br`.

Reply:
That's right, it will match the ` <broken>` tag. I've added that part because a BR tag can contains attributes, and thought we needed to match that case. 
Looking at it now, I think we can simplify the regexp, and make it in par with Aztec. 

Didn't introduce the variable to save memory, but probably overthinking on it. will add a const.